Fix linux-2.6 in download URL for RedHat x86_64

siegy22 opened the pull request:
#### Pull Request (PR) description
Staring with 9.0.5 Splunk changed the filename of the RPM. It no longer includes
"linux-2.6", thus we need to adapt to this change.

```diff
- https://download.splunk.com/products/splunk/releases/9.0.0/linux/splunk-9.0.0-6818ac46f2ec-linux-2.6-x86_64.rpm
+ https://download.splunk.com/products/splunk/releases/9.0.5/linux/splunk-9.0.5-e9494146ae5c.x86_64.rpm
```

#### This Pull Request (PR) fixes the following issues
Replace this comment with the list of issues or n/a.
Use format:
Fixes #348

luedderd commented:
@siegy22 can you retrigger the CI checks by either closing/reopening this PR or doing the following:

```
git commit -m "retrigger checks" --allow-empty
git push
```

#348 is blocking an upgrade deployment in my environment, I'd troubleshoot the failed checks here to help contribute towards a merge but the logs have since expired. Thanks!

siegy22 commented:
@luedderd I amended and force pushed, CI is running once again 😄

luedderd commented:
I haven't worked with these kind of Puppet CI tests before but it seems they're failing in all current PRs for a different issue related to the seed admin password.

@kenyon @bastelfreak seem to be the most active approvers, is there anything else holding this from being approved/merged?  Thanks in advance

siegy22 commented:
@luedderd I spent some hours trying to debug the GitHub actions. Finally found out what's the problem, I opened a separate PR: https://github.com/voxpupuli/puppet-splunk/pull/356 
After this is merged, we can probably rebase and go on here. 🎉

siegy22 commented:
@luedderd I rebased onto my other branch and also added tests. 
As of now, you could set `splunk::enterprise::enterprise_package_src` to the full URL to make it work. E.g. `https://download.splunk.com/products/splunk/releases/9.2.0/linux/splunk-9.2.0-1fff88043d5f.x86_64.rpm` for 9.0.2

bastelfreak commented:
With this change, is it still required that net-tools are explicitly installed?
https://github.com/voxpupuli/puppet-splunk/blob/069a1424f6749fa9a11983331f6016ea08c5d67d/manifests/forwarder/install.pp#L64-L70

that's really legacy tooling and the rpm should depend on it if it's an actual dependency. Do we need to explicitly install it?

siegy22 commented:
@bastelfreak Didn't check that yet. It's possible that this is not needed anymore. But I'd down to check that in a separate branch/PR. We need to ensure that Splunk 9.x can be installed and upgraded. As of now, even 8.2 is EOL and thus we should probably set the focus to be 9.x compatible for the future.

This change just fixes the URL that is being generated to download splunk.

bastelfreak commented:
We have acceptance tests at: https://github.com/voxpupuli/puppet-splunk/tree/master/spec/acceptance

can you add one with the new version so we can verify that it works?
If you don't know beaker: The tests are actual puppet code, executed in different docker containers. duplicate one of the existing blocks and add the correct version.

siegy22 commented:
There seems to be a issue unrelated to this change when installing 9.2.0.1. The package is downloaded and installed correctly (so this change is working) but the default parameters don't make sense with the newer versions. 

I would suggest opening another PR to test the installation and upgrade of 9.x versions on all the platforms. I can do that.

bastelfreak commented:
shouldn't that go into this PR as well? We currently don't support version 9 at all and this PR now implements it in parts, but it's not really usable because of the wrong parameters?

kenyon reviewed (commented):
`net-tools` is actually only required for the Splunk Add-on for Unix and Linux, which collects data on hosts where the splunk forwarder is installed, but I think this add-on is typically always installed on hosts where the forwarder is installed. The docs at https://docs.splunk.com/Documentation/AddOns/released/UnixLinux/Requirements say it's only needed on EL 7 and EL 8, but I wonder if that just hasn't been updated to include EL 9 also.

I wonder if we should just remove everything related to `net-tools` from this module and leave that up to the user, though.

Related: #308

siegy22 commented:
@kenyon I tried it before with a RHEL 9 instance, installing some 8.x versions and 9.x versions. No issues without having `net-tools` installed. The only version that didn't install that I tried was `7.2.4.2` that said net-tools needed to be installed. Maybe they included the "Splunk Add-on for Unix and Linux" before? But it's no longer relevant for 8.x and 9.x as far as I can tell.

siegy22 commented:
@bastelfreak @kenyon Can you review this?

kenyon commented on manifests/forwarder.pp:
```diff
@@ -165,7 +165,10 @@
-  String[1] $splunk_user                     = $splunk::params::splunk_user,
+  String[1] $splunk_user                     = versioncmp($version, '9.1.0') ? {
+    -1 => $splunk::params::splunk_user,
+    default => 'splunkfwd'
```
```suggestion
    default => 'splunkfwd',
```

siegy22 replied:
Added

kenyon commented on manifests/params.pp:
```diff
@@ -112,7 +112,7 @@
   String[1] $splunk_user                     = $facts['os']['family'] ? {
     'windows' => 'Administrator',
-    default => 'root'
+    default => versioncmp($version, '8.0.0') ? { -1 => 'root', default => 'splunk' }
```
```suggestion
    default => versioncmp($version, '8.0.0') ? { -1 => 'root', default => 'splunk' },
```

siegy22 replied:
Added

luedderd commented:
Thanks everyone for working on this! I've tested the forked module and it's worked so far on my handful of testing servers, as soon as this is merged/released I look forward to promoting it to the rest of my node inventory. 🎉
